### ISSUE TITLE
Proxy error: unable to get local issuer certificate

### ISSUE THREAD

**Issue by joaomoreno** [Microsoft/vscode#1862]
From @dogfish182

![2016-01-07_18-35-46](https://cloud.githubusercontent.com/assets/5390419/12177701/d68119de-b56d-11e5-9a9d-d92e1bdca938.png)

```
unable to get local issuer certificate
```

**Comment by joaomoreno** [comment]
Related to nodejs/node#3742

**Comment by dogfish182** [comment]
I'm not sure how to address this for myself as I have a standard install of win10 without any node stuff that i knowingly installed... also the issue presents on a fresh windows 7 corporate image that we don't use for development, so a basic install of win 7 for us (and 10 and 8.1) will all exhibit the same issue.

**Comment by colhountech** [comment]
This worked for me:
start a command prompt and navigate to working folder.

```
set NODE_TLS_REJECT_UNAUTHORIZED=0
code .
```
from http://stackoverflow.com/questions/33820733/error-unable-to-verify-the-first-certificate-how-to-trust-all-certificates

**Comment by joaomoreno** [comment]
@colhountech Does it work if you set `"http.proxyStrictSSL": false` in your settings?

**Comment by wgknowles** [comment]
// Whether the proxy server certificate should be verified against the list of supplied CAs.
This leads me to think there should be a way to add a CA to the "list of supplied CAs".  Ideally, there should be a setting like http.useMachineCAStore that taps into the trusted root CA list of the system (same one used by IE)

Doesn't  setting "http.proxyStrictSSL": false result in all CA's and self-signed certs being trusted?  Seems like we are asking for a MITM hack.

**Comment by psmoot** [comment]
I'm having the same issue on Ubuntu 16 behind a firewall. Setting NODE_TLS_REJECT_UNAUTHORIZED worked but setting http.proxyStrictSSL did not.

I'm happy to try new builds if that will help resolving the issue, just let me know.

**Comment by joaomoreno** [comment]
@psmoot #11702 Fix is out in 1.5.2.

@wgknowles Unfortunately we didn't dedicate time to look into the CA list yet.

Closing this as it seems a duplicate of #11702.
